src/LogOstreamInserters.cpp

```diff
@@ -0,0 +1,200 @@
+ *
+ * @param ostream The output stream.
+ * @param serverConfig The ServerConfig object.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, const ServerConfig& serverConfig)
+{
+	ostream << "Server name: " << serverConfig.serverName << '\n';
+	ostream << "Host: " << serverConfig.host << '\n';
+	ostream << "Port: " << serverConfig.port << '\n';
+	ostream << "Max body size: " << serverConfig.maxBodySize << '\n';
+	ostream << "Error pages:\n";
+	for (std::map<unsigned short, std::string>::const_iterator it = serverConfig.errorPage.begin();
+		 it != serverConfig.errorPage.end(); ++it)
+		ostream << "  " << it->first << ": " << it->second << '\n';
+	ostream << "Locations:\n";
+	for (std::vector<Location>::const_iterator it = serverConfig.locations.begin(); it != serverConfig.locations.end();
+		 ++it)
+		ostream << *it;
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to append a ConfigFile.
+ *
+ * @param ostream The output stream.
+ * @param configFile The ConfigFile object.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, const ConfigFile& configFile)
+{
+	ostream << "Config file:\n";
+	for (std::vector<ServerConfig>::const_iterator it = configFile.serverConfigs.begin();
+		 it != configFile.serverConfigs.end(); ++it)
+		ostream << *it;
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to a Method.
+ *
+ * Translates the Method enum to a string.
+ * If the Method is MethodCount, it will output "enum MethodCount".
+ * @param ostream The output stream.
+ * @param method The Method enum.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, Method method)
+{
+	switch (method) {
+	case MethodGet:
+		ostream << "GET";
+		break;
+	case MethodPost:
+		ostream << "POST";
+		break;
+	case MethodDelete:
+		ostream << "DELETE";
+		break;
+	case MethodCount:
+		ostream << "enum MethodCount";
+		break;
+	}
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to append a statusCode.
+ *
+ * @param ostream The output stream.
+ * @param statusCode The statusCode enum.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, statusCode statusCode)
+{
+	switch (statusCode) {
+	case StatusOK:
+		ostream << "200 OK";
+		break;
+	case StatusMovedPermanently:
+		ostream << "301 Moved Permanently";
+		break;
+	case StatusBadRequest:
+		ostream << "400 Bad Request";
+		break;
+	case StatusForbidden:
+		ostream << "403 Forbidden";
+		break;
+	case StatusNotFound:
+		ostream << "404 Not Found";
+		break;
+	case StatusMethodNotAllowed:
+		ostream << "405 Method Not Allowed";
+		break;
+	case StatusInternalServerError:
+		ostream << "500 Internal Server Error";
+		break;
+	case StatusMethodNotImplemented:
+		ostream << "501 Method Not Implemented";
+		break;
+	case StatusNonSupportedVersion:
+		ostream << "505 HTTP Version Not Supported";
+		break;
+	}
```

Comment:
Same as above, would provide a default case.

Reply:
See above

Reply:
Can be treated as above.